Modules/_abc.c

```diff
@@ -446,6 +449,27 @@ _abc__abc_init(PyObject *module, PyObject *self)
         return NULL;
     }
     Py_DECREF(data);
+    if (PyType_Check(self)) {
+        PyTypeObject *cls = (PyTypeObject *)self;
+        PyObject *flags = _PyDict_GetItemIdWithError(cls->tp_dict, &PyId___flags__);
+        if (flags == NULL) {
+            if (PyErr_Occurred()) {
+                return NULL;
+            }
+        }
+        else {
+            if (PyLong_CheckExact(flags)) {
+                long val = PyLong_AsLong(flags);
+                if (val == -1 && PyErr_Occurred()) {
+                    return NULL;
+                }
+                ((PyTypeObject *)self)->tp_flags |= (val & COLLECTION_FLAGS);
+            }
+            if (_PyDict_DelItemId(cls->tp_dict, &PyId___flags__) < 0) {
+                return NULL;
+            }
```

Comment:
Why are we deleting `__flags__` here? It looks like it will get overwritten anyways:

```py
>>> class C:
...     __flags__ = "Spam"
... 
>>> C.__flags__
284160
```

Reply:
The `__flags__` set in `Sequence` is in `Sequence.__dict__`, but normally `__flags__` is a descriptor.
So, it is probably safest to delete it. I suspect it doesn't matter much in practice.